tests/store/artifact/test_mlflow_artifact_repo.py

```diff
@@ -0,0 +1,300 @@
+import os
+from unittest import mock
+import posixpath
+import pytest
+from urllib.parse import urlparse
+
+from mlflow.store.artifact.mlflow_artifacts_repo import MlflowArtifactsRepository
+from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
+from mlflow.exceptions import MlflowException
+from mlflow.tracking._tracking_service.utils import get_tracking_uri, set_tracking_uri
+
+ORIGINAL_TRACKING_URI = get_tracking_uri()
+
+
+def _set_temp_tracking_uri():
+    temp_uri = "http://localhost:5000/api/2.0/mlflow-artifacts/artifacts"
+    set_tracking_uri(temp_uri)
+    return temp_uri
+
+
+def test_artifact_uri_factory():
+    _set_temp_tracking_uri()
+    repo = get_artifact_repository("mlflow-artifacts://test.com")
+    assert isinstance(repo, MlflowArtifactsRepository)
+    set_tracking_uri(ORIGINAL_TRACKING_URI)
+
+
+def test_mlflow_artifact_uri_formats_resolved():
+    base_api_uri = "api/2.0/mlflow-artifacts/artifacts"
+    uri_temp = _set_temp_tracking_uri()
+    tracking_uri = urlparse(uri_temp)
+    base_path = "/my/artifact/path"
+    conditions = [
+        (
+            f"mlflow-artifacts://myhostname:4242{base_path}/hostport",
+            f"{tracking_uri.scheme}://myhostname:4242/{base_api_uri}{base_path}/hostport",
+        ),
+        (
+            f"mlflow-artifacts://myhostname{base_path}/host",
+            f"{tracking_uri.scheme}://myhostname/{base_api_uri}{base_path}/host",
+        ),
+        (
+            f"mlflow-artifacts:{base_path}/nohost",
+            f"{tracking_uri.scheme}://{tracking_uri.netloc}/{base_api_uri}{base_path}/nohost",
+        ),
+        (
+            f"mlflow-artifacts://{base_path}/redundant",
+            f"{tracking_uri.scheme}://{tracking_uri.netloc}/{base_api_uri}{base_path}/redundant",
+        ),
+        ("mlflow-artifacts:/", uri_temp,),
```

Comment:
```suggestion
        ("mlflow-artifacts:/", uri_temp,),
```

@BenWilson2 Removing the command here should fix the lint issue.